File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/view/CustomTextInputEditText.java
package com.lattis.ellipse.presentation.view;

import android.annotation.TargetApi;
import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Typeface;
import android.os.Build;
import android.util.AttributeSet;

import com.google.android.material.textfield.TextInputEditText;
import com.lattis.ellipse.presentation.view.utils.TypefaceLoader;

import io.lattis.ellipse.R;

public class CustomTextInputEditText extends TextInputEditText {


    public CustomTextInputEditText(Context context) {
        super(context);
    }

    public CustomTextInputEditText(Context context, AttributeSet attrs) {
        super(context, attrs);

        init(context, attrs);
    }

    public CustomTextInputEditText(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }


    private void init(Context context, AttributeSet attrs) {
        TypedArray a = context.obtainStyledAttributes(attrs, R.styleable.CustomTypeFace);
        String font = a.getString(R.styleable.CustomTypeFace_typefaceAsset);

        if (font != null) {
            String path = String.format("fonts/%s", font);
            Typeface tf = TypefaceLoader.getTypeface(context, path);
            if (tf != null) {
                setTypeface(tf);
            }
        }

        a.recycle();
    }
}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/customview/CustomTextInputEditText.kt
package com.lattis.lattis.presentation.customview

import android.content.Context
import android.graphics.Typeface
import android.util.AttributeSet
import com.google.android.material.textfield.TextInputEditText
import com.lattis.lattis.presentation.utils.TypefaceLoader
import io.lattis.lattis.R

class CustomTextInputEditText : TextInputEditText {
    constructor(context: Context) : super(context) {}
    constructor(context: Context, attrs: AttributeSet) : super(
        context,
        attrs
    ) {
        init(context, attrs)
    }

    constructor(
        context: Context,
        attrs: AttributeSet,
        defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(
        context: Context,
        attrs: AttributeSet
    ) {
        val a = context.obtainStyledAttributes(attrs, R.styleable.CustomTypeFace)
        val font = a.getString(R.styleable.CustomTypeFace_typefaceAsset)
        if (font != null) {
            val path = String.format("fonts/%s", font)
            val tf: Typeface = TypefaceLoader.getTypeface(context, path)
            tf?.let { setTypeface(it) }
        }
        a.recycle()
    }
}